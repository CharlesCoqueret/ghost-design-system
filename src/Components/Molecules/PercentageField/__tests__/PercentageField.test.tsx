import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { PercentageField } from '..';

describe('PercentageField Component', () => {
  it('renders properly', () => {
    const { container } = render(<PercentageField name='name' />);
    expect(container).toMatchSnapshot();
  });

  it('renders with values in readonly', () => {
    const { container } = render(<PercentageField input={12.34} name='name' readOnly />);
    expect(container).toMatchSnapshot();
  });

  it('renders with values in disabled highligted', () => {
    const { container } = render(<PercentageField input={12345.67} name='name' disabled highlighted />);
    expect(container).toMatchSnapshot();
  });

  it('renders with values with fieldSize and inline', () => {
    const { container } = render(<PercentageField input={1.23} name='name' inline fieldSize={6} />);
    expect(container).toMatchSnapshot();
  });

  it('renders with values set as string', () => {
    const { container } = render(<PercentageField input={'1.23'} name='name' />);
    expect(container).toMatchSnapshot();
  });

  it('renders with values set as undefined', () => {
    const { container } = render(<PercentageField input={undefined} name='name' />);
    expect(container).toMatchSnapshot();
  });

  it('does nothing when onChange is not defined', async () => {
    const { container } = render(<PercentageField input={1.23} name='name' dataTestId='DATA-TEST-ID' />);
    expect(container).toMatchSnapshot();

    const inputNode = screen.getByTestId('DATA-TEST-ID');

    await userEvent.type(inputNode, '{backspace}');

    expect(container).toMatchSnapshot();
  });

  it('handles change', async () => {
    const onChangeMock = jest.fn().mockImplementation(() => {});

    const { container } = render(
      <PercentageField input={1.23} name='name' onChange={onChangeMock} dataTestId='DATA-TEST-ID' />,
    );
    expect(container).toMatchSnapshot();

    const inputNode = screen.getByTestId('DATA-TEST-ID');

    await userEvent.type(inputNode, '{backspace}');

    expect(onChangeMock).toBeCalledTimes(1);
    expect(onChangeMock).toHaveBeenLastCalledWith(0.12);
    expect(container).toMatchSnapshot();
  });
});
