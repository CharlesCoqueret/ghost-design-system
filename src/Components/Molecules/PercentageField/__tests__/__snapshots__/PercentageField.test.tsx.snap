// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PercentageField Component does nothing when onChange is not defined 1`] = `
<div>
  <div
    class="fieldGroup"
  >
    <input
      autocomplete="off"
      class="default"
      data-testid="DATA-TEST-ID"
      inputmode="numeric"
      name="name"
      type="text"
      value="123 %"
    />
  </div>
</div>
`;

exports[`PercentageField Component does nothing when onChange is not defined 2`] = `
<div>
  <div
    class="fieldGroup"
  >
    <input
      autocomplete="off"
      class="default"
      data-testid="DATA-TEST-ID"
      inputmode="numeric"
      name="name"
      type="text"
      value="12 %"
    />
  </div>
</div>
`;

exports[`PercentageField Component handles change 1`] = `
<div>
  <div
    class="fieldGroup"
  >
    <input
      autocomplete="off"
      class="default"
      data-testid="DATA-TEST-ID"
      inputmode="numeric"
      name="name"
      type="text"
      value="123 %"
    />
  </div>
</div>
`;

exports[`PercentageField Component handles change 2`] = `
<div>
  <div
    class="fieldGroup"
  >
    <input
      autocomplete="off"
      class="default"
      data-testid="DATA-TEST-ID"
      inputmode="numeric"
      name="name"
      type="text"
      value="12 %"
    />
  </div>
</div>
`;

exports[`PercentageField Component renders properly 1`] = `
<div>
  <div
    class="fieldGroup"
  >
    <input
      autocomplete="off"
      class="default"
      inputmode="numeric"
      name="name"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`PercentageField Component renders with values in disabled highligted 1`] = `
<div>
  <div
    class="fieldGroup"
  >
    <input
      autocomplete="off"
      class="default disabled"
      disabled=""
      inputmode="numeric"
      name="name"
      type="text"
      value="1,234,567 %"
    />
  </div>
</div>
`;

exports[`PercentageField Component renders with values in readonly 1`] = `
<div>
  <div
    class="fieldGroup"
  >
    <span
      autocomplete="off"
      class="default readOnly"
      id="name"
      name="name"
    >
      1,234 %
    </span>
  </div>
</div>
`;

exports[`PercentageField Component renders with values set as string 1`] = `
<div>
  <div
    class="fieldGroup"
  >
    <input
      autocomplete="off"
      class="default"
      inputmode="numeric"
      name="name"
      type="text"
      value="123 %"
    />
  </div>
</div>
`;

exports[`PercentageField Component renders with values set as undefined 1`] = `
<div>
  <div
    class="fieldGroup"
  >
    <input
      autocomplete="off"
      class="default"
      inputmode="numeric"
      name="name"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`PercentageField Component renders with values with fieldSize and inline 1`] = `
<div>
  <div
    class="fieldGroup size-6 inline"
  >
    <input
      autocomplete="off"
      class="default"
      inputmode="numeric"
      name="name"
      type="text"
      value="123 %"
    />
  </div>
</div>
`;
